src/targets/__tests__/npm.test.ts

```diff
@@ -0,0 +1,129 @@
+import { parseVersion } from '../../utils/version';
+import { getPublishTag, getLatestVersion } from '../npm';
+
+function setNpmToken() {
+  process.env.NPM_TOKEN = 'test-token';
+}
+
+const defaultNpmConfig = {
+  useYarn: false,
+  token: 'xxx',
+};
+
+describe('getLatestVersion', () => {
+  beforeEach(() => {
+    setNpmToken();
+  });
+
+  it('returns undefined for unexisting checkPackageName', async () => {
+    const actual = await getLatestVersion(
+      'sentry-xx-this-does-not-exist',
+      defaultNpmConfig
+    );
+    expect(actual).toEqual(undefined);
+  });
+
+  it('returns version for valid id', async () => {
+    const actual = await getLatestVersion('@sentry/browser', defaultNpmConfig);
+    expect(actual).toBeDefined();
+
+    // Returns a valid version
+    const parsed = parseVersion(actual as string);
+    expect(parsed).toBeDefined();
+    expect(parsed?.major).toBeGreaterThanOrEqual(7);
+  });
+});
+
+describe('getPublishTag', () => {
+  beforeEach(() => {
+    setNpmToken();
+  });
+
+  it('returns undefined without a checkPackageName', async () => {
+    const logger = {
+      warn: jest.fn(),
+    } as any;
+    const actual = await getPublishTag(
```

Comment:
hmmm is this hitting actual public npm during test? we probably want to avoid that if possible since it makes these tests dependent on external state

Reply:
yes, this hits npm. I tried to write the test in a way that it works with whatever version we get back. But yes, if NPM would be down this would fail. Any idea how else to test this? 🤔 if we don't want to hit NPM the test also gets less realistic. The only thing I can think of is to mock `spawnProcess` and have it return a static version?

Reply:
@asottile-sentry I've updated the test to use a mocked process there instead!